libs/generic-view/store/src/lib/toasts/open-toast.action.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { createAsyncThunk } from "@reduxjs/toolkit"
+import { Toast } from "./reducer"
+import { ActionName } from "../action-names"
+import { ReduxRootState } from "Core/__deprecated__/renderer/store"
+import { addToast } from "./actions"
+
+export const openToastAction = createAsyncThunk<
+  void,
+  Toast["key"],
+  { state: ReduxRootState }
+>(ActionName.OpenToast, async (toastKey, { dispatch }) => {
+  dispatch(addToast(toastKey))
+  // Delay action to add toast before the state will update
+  await new Promise((resolve) => setTimeout(resolve, 1))
```

Comment:
We want to show the number of deleted contacts in the toast and this value is taken from `selectedContacts` form property. The issue is that deleting contacts immediately deselects them which would lead to showing "0 contacts deleted" in the toast. To prevent this, the deselection must happen in the next cycle so the value inside the toast can be properly read before the `selectedContacts` will update.